commands/createfixpullrequests.go

```diff
@@ -95,113 +104,186 @@ func (cfp *CreateFixPullRequestsCmd) scan(scanSetup *utils.ScanDetails, currentW
-	wd, err := fileutils.CreateTempDir()
+func (cfp *CreateFixPullRequestsCmd) fixVulnerablePackages(fixVersionsMap map[string]*FixVersionInfo) (err error) {
+	cfp.gitManager, err = utils.NewGitManager(cfp.dryRun, cfp.dryRunRepoPath, ".", "origin", cfp.details.Token, cfp.details.Username)
+	if err != nil {
+		return err
+	}
+
+	clonedRepoDir, restoreBaseDir, err := cfp.cloneRepository()
 	if err != nil {
 		return err
 	}
 	defer func() {
-		e := fileutils.RemoveTempDir(wd)
-		if err == nil {
-			err = e
+		e1 := restoreBaseDir()
+		e2 := fileutils.RemoveTempDir(clonedRepoDir)
+		if err == nil && (e1 != nil || e2 != nil) {
+			err = fmt.Errorf("%s\n%s", e1, e2)
 		}
 	}()
-	log.Debug("Created temp working directory:", wd)
 
-	// Clone the content of the repo to the new working directory
-	gitManager, err := utils.NewGitManager(cfp.dryRun, cfp.dryRunRepoPath, ".", "origin", repoGitParams.Token, repoGitParams.Username)
+	// Fix all impacted packages
+	for impactedPackage, fixVersionInfo := range fixVersionsMap {
+		if err = cfp.fixSinglePackage(impactedPackage, fixVersionInfo); err != nil {
+			log.Warn(err)
+		}
+		// After finishing to work on the current vulnerability we go back to the base branch to start the next vulnerability fix
+		log.Info("Running git checkout to base branch:", cfp.details.Branch)
+		if err = cfp.gitManager.Checkout(cfp.details.Branch); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (cfp *CreateFixPullRequestsCmd) fixSinglePackage(impactedPackage string, fixVersionInfo *FixVersionInfo) (err error) {
+	log.Info("-----------------------------------------------------------------")
+	log.Info("Start fixing", impactedPackage, "with", fixVersionInfo.fixVersion)
+	fixBranchName, err := cfp.createFixingBranch(impactedPackage, fixVersionInfo)
+	if err != nil {
+		return fmt.Errorf("failed while trying to create new branch: \n%s", err.Error())
+	}
+
+	if err = cfp.updatePackageToFixedVersion(fixVersionInfo.packageType, impactedPackage, fixVersionInfo.fixVersion); err != nil {
+		return fmt.Errorf("failed while trying to fix %s with version: %s with error: \n%s", impactedPackage, fixVersionInfo.fixVersion, err.Error())
+	}
+
+	if err = cfp.openFixingPullRequest(impactedPackage, fixBranchName, fixVersionInfo); err != nil {
+		return fmt.Errorf("failed while trying to create a fixing pull request for: %s with version: %s with error: \n%s",
+			impactedPackage, fixVersionInfo.fixVersion, err.Error())
+	}
+	return
+}
+
+func (cfp *CreateFixPullRequestsCmd) openFixingPullRequest(impactedPackage, fixBranchName string, fixVersionInfo *FixVersionInfo) (err error) {
+	log.Info("Checking if there are changes to commit")
+	isClean, err := cfp.gitManager.IsClean()
 	if err != nil {
 		return err
 	}
+	if isClean {
+		return fmt.Errorf("there were no changes to commit after fixing the package '%s'", impactedPackage)
+	}
 
-	err = gitManager.Clone(wd, branch)
+	log.Info("Running git add all and commit")
```

Comment:
Running git add all and commit --> Running git add all and commit...

Also, let's place this line right before -
```
err = cfp.gitManager.AddAllAndCommit(commitString)
```

Reply:
The sentence wasn't changed, what did you mean?

Reply:
done